src/biotite/structure/io/pdbx/convert.py

```diff
@@ -1658,9 +1658,13 @@ def _apply_transformations(structure, transformation_dict, operations):
             coord = matrix_rotate(coord, rotation_matrix)
             # Translate
             coord += translation_vector
+
+        sym_ids.append("-".join(list(operation)))
         assembly_coord[i] = coord
 
-    return repeat(structure, assembly_coord)
+    assembly = repeat(structure, assembly_coord)
+    assembly.set_annotation("sym_id", np.repeat(sym_ids, structure.array_length()))
```

Comment:
@padix-key @alex-hh 
Sorry for taking longer to get back -- and thank you for this, this is very helpful!

**TL;DR**: I still believe an integer encoding would be a better design decision here that'd be just as easy to implement, but agree that this is a nit-picky point that may only be relevant in a handful of cases. Overall I'd be happy with the current implementation.

- `4v5x`: you're right - bad choice by me this one indeed only has one assembly (but a ton of chains :D) 
- `6o3h`: Thank you for going in depth with this example. I agree that `"60"` would not be super concerning. 
- `1f2n`: This was the type of behaviour I was worried about, thank you for identifying one of these cases. For this particular one, the string only gets up to length `"X0-60"` , which isn't terrible but still pretty hefty compared to what it could be: stored e.g. via numpy's `<U` encoding -- utf32, i.e. 4 bytes iirc -- this would be `5*4=20` bytes per element. That's more than the `3*4=12` (float 32) bytes per coordinate in this case. Instead, with an integer encoding this could easily be `1` byte (uint8) or `2` bytes (uint16). This would make the atom-array both, more lightweight and comparisons over the `sym_id` dimension faster because we can do int comparisons rather than string comparisons.

What I don't know is if there are cases where this id could become even longer? (@padix-key or @nscorley, maybe you know more?). In the past I've often been surprised by obscure edge cases in the PDB that I thought would never happen, so nowadays I usually try to rely on as little assumptions on PDB files as possible.

Reply:
No I have no preference - I was just aiming for consistency with the PDB. (I also wanted to check that the examples you gave were in fact consistent).

I guess one consideration would be that in most cases the string will be an integer and it makes sense for this integer to be the same as the one used to identify the transformation in the cif file.

In cases where we compose transformations (it would be cool to have a proper complex example but I don't) I have no preference and don't even know what the PDB convention is.

Reply:
Okay, then I would settle with the integer counting up from `0` to `N`. I would also consistently always do it this way, even if the transformation is not composed of multiple operations, so the user knows what to expect.

@alex-hh Would you like to add the necessary changes, or should I do it?

Reply:
Sounds good: if you're happy to do it it might be quicker! But also happy to if you don't have time

Reply:
> @padix-key @alex-hh Sorry for taking longer to get back -- and thank you for this, this is very helpful!
> 
> **TL;DR**: I still believe an integer encoding would be a better design decision here that'd be just as easy to implement, but agree that this is a nit-picky point that may only be relevant in a handful of cases. Overall I'd be happy with the current implementation.
> 
> * `4v5x`: you're right - bad choice by me this one indeed only has one assembly (but a ton of chains :D)
> * `6o3h`: Thank you for going in depth with this example. I agree that `"60"` would not be super concerning.
> * `1f2n`: This was the type of behaviour I was worried about, thank you for identifying one of these cases. For this particular one, the string only gets up to length `"X0-60"` , which isn't terrible but still pretty hefty compared to what it could be: stored e.g. via numpy's `<U` encoding -- utf32, i.e. 4 bytes iirc -- this would be `5*4=20` bytes per element. That's more than the `3*4=12` (float 32) bytes per coordinate in this case. Instead, with an integer encoding this could easily be `1` byte (uint8) or `2` bytes (uint16). This would make the atom-array both, more lightweight and comparisons over the `sym_id` dimension faster because we can do int comparisons rather than string comparisons.
> 
> What I don't know is if there are cases where this id could become even longer? (@padix-key or @nscorley, maybe you know more?). In the past I've often been surprised by obscure edge cases in the PDB that I thought would never happen, so nowadays I usually try to rely on as little assumptions on PDB files as possible.

@Croydon-Brixton In my view, either approach would work, but I tend to prefer the integer solution as well, for both the reasons that you and @padix-key suggested.

The longest symmetry ID I've seen (and thus a lower bound) is in `2fs3`: `'(X0)(1-10,21-25,31-35,41-50)'`. Given the number of atoms involved in this structure, when building the full biological assembly of up to a million atoms, such a field (a 23-character string) becomes non-negligible (back-of-the-envelope math gives me ~25mb, which is a bit on the larger side, especially given how lightweight the other annotation categories are).

Although converting to integers comes as the cost of losing the information about what transformations were applied to the corresponding chain in the asymmetric unit, I haven't come across any cases where that's needed for downstream uses.

Reply:
Actually the field would not be that long: `X0` would only be combined with one of the other operations, i.e. the sym IDs would be
```
X0-1, ..., X0-10, X0-21, ...
```
Nevertheless, as discussed above, using integers has still other advantages